### PR TITLE
Combine form entry view model factories and convert QuitFormDialogFragment to simple dialog

### PR DESCRIPTION
**Pull request by seadowg**
Follow up to #5366 (discussion [here](https://github.com/getodk/collect/pull/5366#discussion_r1032477176) and [here](https://github.com/getodk/collect/pull/5366#issuecomment-1330615718))

#### What has been done to verify that this works as intended?

New and existing tests and verified manually.

#### Why is this the best possible solution? Were any other approaches considered?

Discussed in #5366 

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

This shouldn't change anything, but a lot of form entry code has been touched. I'd recommend checking form entry in general, but also:

* deleting repeats
* recording audio (for questions and in the background)
* select one from map questions

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

File: androidshared/src/main/java/org/odk/collect/androidshared/ui/DialogService.kt [hunk 1]
```diff
@@ -0,0 +1,109 @@
+package org.odk.collect.androidshared.ui
+
+import android.content.Context
+import android.view.View
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+
+object DialogService {
```
Comment (seadowg) [inline androidshared/src/main/java/org/odk/collect/androidshared/ui/DialogService.kt]: @grzesiek2010 I'd be interested to know how you feel about using our own machinery to fake AndroidX's `AlertDialog` as opposed to using a custom Shadow like I do in https://github.com/getodk/collect/pull/5391/commits/fa5116e3405cc1502eb04e8e9adb6f99c963dae3. There's definitely some more clean up on `DialogService` to be done, but I'd like to get input on the general approach before moving forward as I'm not feeling particularly confident about whether it's better than using Robolectric or not.

Comment (grzesiek2010) [reply androidshared/src/main/java/org/odk/collect/androidshared/ui/DialogService.kt]: The second approach looks much more natural to me but if you feel like that `our own machinery to fake AndroidX's AlertDialog` makes life/testing easier I won't be against it.

Comment (seadowg) [reply androidshared/src/main/java/org/odk/collect/androidshared/ui/DialogService.kt]: Great, thanks! I can see advantages and disadvantages to both, so if you feel like using a custom Shadow is better, let's go with that.

**Review by grzesiek2010 — APPROVED** [review]

**Comment by dbemke** [comment]
**Tested with Success!**

Verified on Android 10

**Verified Cases:**
      - switching to a different app during: Save/Ignore dialog, Permission dialogs, Hierarchy view
      - background processing set to "No"
      - don't keep activities enabled/disabled
      - minimize the app, rotate the screen, lock the screen
      - Select one map widgets
      - restoring to the quit form dialog shows the hierarchy
      - restoring to the hierarchy shows normal form entry instead of the hierarchy
      - restoring in Edit Saved Form, View Sent Form
      - regression checks in Restoring the app from the background when Form Entry dialogs displayed
      - regression checks in forms with audio

**Comment by srujner** [comment]
**Tested with Success!**

Verified on Android 12 and 13

**Pull request merged**